framework/docs/source/how-to-authenticate-supernodes.rst

```diff
@@ -2,90 +2,115 @@ Authenticate SuperNodes
+To establish an authentication mechanism by which only authorized SuperNodes can connect
+to a running SuperLink, a set of key pairs for both SuperLink and SuperNodes need to be
+created.
+
+We have prepared a script that can be used to generate such set of keys. While using
+these are fine for prototyping, we advice you to follow the standards set in your
+team/organization and generated the keys and share them with the corresponding parties.
+Refer to the **Generate public and private keys for SuperNode authentication** section
+in the example linked at the top of this guide.
 
 .. code-block:: bash
 
-    flower-superlink
-        --ssl-ca-certfile certificates/ca.crt
-        --ssl-certfile certificates/server.pem
-        --ssl-keyfile certificates/server.key
-        --auth-list-public-keys keys/client_public_keys.csv
-        --auth-superlink-private-key keys/server_credentials
-        --auth-superlink-public-key keys/server_credentials.pub
+    # In the example directory, generate the public/private key pairs
+    $ ./generate_auth_keys.sh
 
-Let's break down the authentication flags:
+This will generate the keys in a new ``keys/`` directory. By default it creates a key
+pair for the SuperLink and one for each SuperNode. Copy this directory into the
+directory of your app (e.g. a directory generated earlier via ``flwr new``).
 
-1. The first flag ``--auth-list-public-keys`` expects a path to a CSV file storing all
-   known node public keys. You need to store all known node public keys that are allowed
-   to participate in a federation in one CSV file (``.csv``).
+Enable node authentication in SuperLink
+---------------------------------------
 
-       A valid CSV file storing known node public keys should list the keys in OpenSSH
-       format, separated by commas and without any comments. For an example, refer to
-       our code sample, which contains a CSV file with two known node public keys.
+To launch a SuperLink with SuperNode authentication enabled, you need to provide three
+aditional files in addition to the certificates needed for the TLS connections. Recall
+that the authentication feature can only be enabled in the presence of TLS.
 
-2. The second and third flags ``--auth-superlink-private-key`` and
-   ``--auth-superlink-public-key`` expect paths to the server's private and public keys.
-   For development purposes, you can generate a private and public key pair using
-   ``ssh-keygen -t ecdsa -b 384``.
+.. code-block:: bash
+    :emphasize-lines: 5,6,7
+
+    flower-superlink \
+        --ssl-ca-certfile certificates/ca.crt \
+        --ssl-certfile certificates/server.pem \
+        --ssl-keyfile certificates/server.key \
+        --auth-list-public-keys keys/client_public_keys.csv \
+        --auth-superlink-private-key keys/server_credentials \
+        --auth-superlink-public-key keys/server_credentials.pub
 
-.. note::
+.. dropdown:: Understand the command
```

Comment:
The dropdown is a cool idea but it doesn't show properly in Github (see screenshot below).

<img width="1007" alt="Screenshot 2025-01-14 at 23 06 15" src="https://github.com/user-attachments/assets/8c902d0e-d3ea-4f4c-a979-1281d4b4a1bf" />